### PR TITLE
Make s3_scrubber for safekeepers useful.

### PR DESCRIPTION
**Pull request by arssher**
I'll clean up things a bit more, but consistency checker works and idea is reviewable.

### CODE REVIEW

File: s3_scrubber/src/cloud_admin_api.rs [hunk 1]
```diff
@@ -261,42 +272,34 @@ impl CloudAdminApiClient {
...
-            })?;
+            let response_bytes = CloudAdminApiClient::with_retries(
+                || async {
+                    let response = self
+                        .http_client
+                        .get(self.append_url("/projects"))
+                        .query(&[
+                            ("show_deleted", "false".to_string()),
+                            ("limit", format!("{PAGINATION_LIMIT}")),
+                            ("offset", format!("{pagination_offset}")),
```
Comment (petuhovskiy) [inline s3_scrubber/src/cloud_admin_api.rs:283]: This is not the right way to fetch all projects, offset pagination can have artifacts if someone creates/deletes projects at the time of fetching.

Comment (arssher) [reply s3_scrubber/src/cloud_admin_api.rs:283]: Yeah, but we recheck project later if it is not found in the cache.

File: s3_scrubber/README.md [hunk 2]
```diff
@@ -67,8 +67,10 @@ the purge command will log all the keys that it would have deleted.
 
 #### `scan-metadata`
 
-Walk objects in a pageserver S3 bucket, and report statistics on the contents.
+Walk objects in a pageserver or safekeeper S3 bucket, and report statistics on the contents and checking consistency.
+Errors are logged to stderr and summary to stdout.
 
+For pageserver:
 ```
 env SSO_ACCOUNT_ID=123456 REGION=eu-west-1 BUCKET=my-dev-bucket CLOUD_ADMIN_API_TOKEN=${NEON_CLOUD_ADMIN_API_STAGING_KEY} CLOUD_ADMIN_API_URL=[url] cargo run --release -- scan-metadata
```
Comment (petuhovskiy) [inline s3_scrubber/README.md:75]: Should add `--node-kind pageserver` to the suffix

File: s3_scrubber/src/cloud_admin_api.rs [hunk 3]
```diff
@@ -335,43 +339,61 @@ impl CloudAdminApiClient {
...
+            "find_timeline_branch",
+        )
+        .await?;
+
+        let mut branches: Vec<BranchData> = response.data.into_iter().collect();
+        // Normally timeline_id is unique. However, we do have at least one case
+        // of the same timeline_id in two different projects, apparently after
+        // manual recovery. So always recheck project_id (discovered through
+        // tenant_id).
+        let project_data = match self.find_tenant_project(tenant_id).await? {
```
Comment (petuhovskiy) [inline s3_scrubber/src/cloud_admin_api.rs:380]: nit: this extra request doesn't feel good, as it is not required for 99% of timelines. This is probably not an issue at all, just wanted to mention.

**Review by petuhovskiy — APPROVED** [review]
LGTM

File: s3_scrubber/README.md (hunk 2 above)
Comment (arssher) [reply s3_scrubber/README.md:75]: done

**Comment by arssher** [comment]
Rebased, will merge soon.

**Pull request merged**